lariat_agents/agent/s3_trigger/s3_trigger_agent.py

```diff
@@ -0,0 +1,97 @@
+from lariat_agents.base.base_agent import BaseAgent
+from boto3_type_annotations import s3
+import re
+import boto3
+import json
+from genson import SchemaBuilder
+from lariat_python_common.json.utils import get_clean_schema
+
+
+class S3TriggerAgent(BaseAgent):
+    def __init__(
+        self,
+        agent_type: str,
+        cloud: str,
+        s3_handler: s3.Client,
+        api_key: str = None,
+        application_key: str = None,
+    ):
+        self.s3_handler = s3_handler
+        super().__init__(
+            agent_type=agent_type,
+            cloud=cloud,
+            api_key=api_key,
+            application_key=application_key,
+            query_builder=None,
+        )
+
+    def schema_retrieval(self, event_dict):
+        output_schema_map = {}
+        agent_config = self.get_yaml_config()
+        if "Records" in event_dict:
+            for record in event_dict["Records"]:
+                if "s3" in record:
+                    s3_record = record["s3"]
+                    bucket_name = s3_record["bucket"]["name"]
+                    object_key = s3_record["object"]["key"]
+                    if bucket_name in agent_config["buckets"]:
+                        bucket_configs = agent_config["buckets"][bucket_name]
+                        for bucket_config in bucket_configs:
+                            if object_key.startswith(bucket_config.get("prefix")): # Matches prefix
+                                suffix_key = object_key[len(bucket_config.get("prefix")):]
```

Comment:
Factor this part out to common. Thinking on it, it might be useful in a gcp / azure context (or in other lambdas etc.)